src/code/UNOClasses/Deck.java

```diff
@@ -55,4 +56,34 @@ public void shuffleDeck() {
     public int deckTotal() {
         return deck.size();
     }
-}
+
+    public UNOCard deal() {
+        return deck.remove(0);
+    }
+
+    public String toString() {
+        return deck.stream().map(b -> b.toString()).collect(Collectors.joining("\n"));
+    }
+
+    public boolean isEmpty() {
+        return this.deck.isEmpty();
+    }
+
+    public void shuffleDiscardPile(Vector<UNOCard> cards) {
+        this.deck.addAll(cards);
```

Comment:
Don't we want to leave the top card on the discard pile so the next person can continue playing the game?

Reply:
I think that Game will handle popping off the top card before sending the rest to the shuffleDiscardPile function.

Reply:
Ok that works for me. As long as we consider that :)

Reply:
Because we are keeping the discard pile in Game, it makes sense for the method to be there to pop off the top.